qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -168,30 +227,56 @@ def load_account(self) -> Optional[AccountProvider]:
                            'account.')
             return None
 
-        return providers[0]
+        # The provider for the default open access project.
+        default_provider = providers[0]
+
+        # If specified, attempt to get the provider stored for the account.
+        if stored_provider_hgp:
+            hub, group, project = stored_provider_hgp.to_tuple()
+            try:
+                default_provider = self.get_provider(hub=hub, group=group, project=project)
+            except IBMQProviderError as ex:
+                raise IBMQAccountError('The default provider (hub/group/project) stored on '
+                                       'disk could not be found: '
+                                       'hub = "{}", group = "{}", project = "{}"'
```

Comment:
It's probably helpful to tell the users that they can overwrite the default provider stored on the disk with `save_account(overwrite=True)` in the error message.